app/test_main.py

```diff
@@ -2,0 +1,77 @@
+from typing import Callable
+
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.fixture()
+def mocked_valid_url() -> None:
+    with mock.patch("app.main.valid_google_url") as mock_valid_url:
+        yield mock_valid_url
+
+
+@pytest.fixture()
+def mocked_has_connection() -> None:
+    with mock.patch("app.main.has_internet_connection") as mock_has_connection:
+        yield mock_has_connection
+
```

Comment:
The name of the fixture `mocked_has_connection` is not descriptive enough. It would be better to rename it to `mock_has_internet_connection` to match the function it's mocking.